### PR TITLE
Add Areatriggers_template to SQL parsing

### PR DESCRIPTION
**Pull request by Traesh**
Related to https://github.com/TrinityCore/TrinityCore/pull/18035

Allow to output SQL for Areatrigger_template & Areatrigger_template_polygon_vertices

### CODE REVIEW

File: WowPacketParser/Store/Objects/AreaTriggerTemplate.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
+﻿using WowPacketParser.Enums;
+using WowPacketParser.Misc;
+using WowPacketParser.SQL;
+
+public enum AreaTriggerFlags : int
```
Comment (Vincent-Michael) [inline WowPacketParser/Store/Objects/AreaTriggerTemplate.cs]: move flags to enums

File: WowPacketParser/SQL/Builders/Miscellaneous.cs [hunk 2]
```diff
@@ -67,6 +67,36 @@ public static string SniffData()
             return SQLUtil.Compare(Storage.SniffData, templateDb, x => string.Empty);
         }
 
+        [BuilderMethod]
+        public static string AreaTriggerTemplateData()
+        {
+            if (Storage.AreaTriggerTemplates.IsEmpty())
+                return string.Empty;
+
+            if (Settings.DumpFormat != DumpFormatType.SniffDataOnly)
```
Comment (Vincent-Michael) [inline WowPacketParser/SQL/Builders/Miscellaneous.cs]: why?

File: WowPacketParser/SQL/Builders/Miscellaneous.cs [hunk 3]
```diff
@@ -67,6 +67,36 @@ public static string SniffData()
...
+            return SQLUtil.Compare(Storage.AreaTriggerTemplates, templateDb, x => string.Empty);
+        }
+
+        [BuilderMethod]
+        public static string AreaTriggerTemplateVerticesData()
+        {
+            if (Storage.AreaTriggerTemplates.IsEmpty())
+                return string.Empty;
+
+            if (Settings.DumpFormat != DumpFormatType.SniffDataOnly)
```
Comment (Vincent-Michael) [inline WowPacketParser/SQL/Builders/Miscellaneous.cs]: same

File: WowPacketParser/SQL/Builders/Miscellaneous.cs (hunk 2 above)
Comment (Traesh) [inline WowPacketParser/SQL/Builders/Miscellaneous.cs]: To please the copy/paste fail god

File: WowPacketParser/SQL/Builders/Miscellaneous.cs (hunk 3 above)
Comment (Vincent-Michael) [inline WowPacketParser/SQL/Builders/Miscellaneous.cs]: same

File: WowPacketParserModule.V7_0_3_22248/Parsers/UpdateHandler.cs [hunk 4]
```diff
@@ -364,99 +364,136 @@ private static MovementInfo ReadMovementUpdateBlock(Packet packet, WowGuid guid,
...
+                    List<AreaTriggerTemplateVertices> verticesList = new List<AreaTriggerTemplateVertices>();
+
+                    areaTriggerTemplate.Data[0] = packet.ReadSingle("Height", index);
+                    areaTriggerTemplate.Data[1] = packet.ReadSingle("HeightTarget", index);
+
+                    for (uint i = 0; i < verticesCount; ++i)
+                    {
+                        AreaTriggerTemplateVertices areaTriggerTemplateVertices = new AreaTriggerTemplateVertices
+                        {
+                            AreaTriggerId = guid.GetEntry(),
```
Comment (Shauren) [inline WowPacketParserModule.V7_0_3_22248/Parsers/UpdateHandler.cs]: Does guid.GetEntry() match OBJECT_FIELD_ENTRY for areatriggers?

Comment (Traesh) [inline WowPacketParserModule.V7_0_3_22248/Parsers/UpdateHandler.cs]: Yes

File: WowPacketParserModule.V7_0_3_22248/Parsers/UpdateHandler.cs [hunk 5]
```diff
@@ -364,99 +364,136 @@ private static MovementInfo ReadMovementUpdateBlock(Packet packet, WowGuid guid,
...
+                        verticesList.Add(areaTriggerTemplateVertices);
+                    }
 
                     for (var i = 0; i < verticesTargetCount; ++i)
-                        packet.ReadVector2("VerticesTarget", index, i);
+                    {
+                        Vector2 verticesTarget = packet.ReadVector2("VerticesTarget", index, i);
+
+                        verticesList[i].VerticeTargetX = verticesTarget.X;
+                        verticesList[i].VerticeTargetY = verticesTarget.Y;
```
Comment (joschiwald) [inline WowPacketParserModule.V7_0_3_22248/Parsers/UpdateHandler.cs]: VerticeTarget is not passed properly to output, always UNKNOWN in sql output

**Comment by funjoker** [comment]
any news?

**Comment by Traesh** [comment]
I'll wait for the TC PR to be merged to finalize this one, it will avoid doing things twice if areatrigger_* DB struct change again

**Pull request merged**